tests/dataloader_tests.cpp

```diff
@@ -58,11 +60,28 @@ BOOST_AUTO_TEST_CASE(CSVDataLoaderTest)
 BOOST_AUTO_TEST_CASE(MNISTDataLoaderTest)
 {
   DataLoader<> dataloader("mnist", true, 0.80);
+
   // Check for correct dimensions.
-  BOOST_REQUIRE_EQUAL(dataloader.TrainFeatures().n_cols, 784);
-  BOOST_REQUIRE_EQUAL(dataloader.TestFeatures().n_cols, 784);
-  BOOST_REQUIRE_EQUAL(dataloader.ValidFeatures().n_cols, 784);
-  BOOST_REQUIRE_EQUAL(dataloader.TrainFeatures().n_rows, 33600);
+  BOOST_REQUIRE_EQUAL(dataloader.TrainFeatures().n_rows, 784);
+  BOOST_REQUIRE_EQUAL(dataloader.TestFeatures().n_rows, 784);
+  BOOST_REQUIRE_EQUAL(dataloader.ValidFeatures().n_rows, 784);
+
+
```

Comment:
Can you use 1 newline and add explanation about what we're checking in this block?

And, perhaps, I missed something... can you let me know why the data were distributed like below?
`Train : 8,400`
`Valid: 33,600`
`Test: 28,000`

Reply:
This is the same split as mlpack::data::split. Here 0.8 means 80% testing and 20 % training data. mlpack takes in test ratio. I though it would be best not to invert. Let me know if I should do that or keep it same as mlpack.

Reply:
Ok, then, it would be better to change the parameter name from `ratio` to `testRatio` for clarification :)

Reply:
Sure, I'll make the change.

Reply:
Also, kindly update the parameter description as well in [here](https://github.com/kartikdutt18/models/blob/6ae432e2951f2a222fc37fe796b7131dea0e9da1/dataloader/dataloader.hpp#L88)

Reply:
Sure, will make the changes. Thanks a lot.

Reply:
It seems [this](https://github.com/kartikdutt18/models/blob/6ae432e2951f2a222fc37fe796b7131dea0e9da1/dataloader/dataloader.hpp#L109) also needs to be changed

Reply:
I think I can set it to 0.25 instead. I think that makes sense.

Reply:
Yes, but I think the `ratio` is a little vague, so it should be changed to `testRatio` and description is also needed to be clear that is a ratio for test data.

Reply:
Ah, but it's not the work for this PR... sorry. I can do it.

Reply:
I can make the change. I almost done with this commit.

Reply:
Done, the tests pass locally. The build should reciprocate the same. Opening the other PR now.

Reply:
I probably missed something, the mlpack::data::split already uses testRatio and I changed it here too. Kindly let me know if I missed something.

Reply:
I requested the changes because I think the parameter name `ratio` in `DataLoader` is passed to `mlpack::data::Split()` that uses `testRatio` and the name `ratio` in `DataLoader` is a little vague because I seems train-test split ratio, not test-train ratio. Please let me know if it doesn't make sense.

Also, I'm still trying to find why the data distribution ratio is.
```
Train : 8,400 (12%)
Valid: 33,600 (48%)
Test: 28,000 (40%)
```

The total number of data is `70,000` and if we split the whole data with the test ratio `0.8`, the train-test ratio should be `0.2` and `0.8`. And the current `DataLoader` implementation splits the train data into train and valid data using the test ratio.

So, shouldn't the data ratio be like this? Are the sources of train and test data different?
``
Train: 4%
Valid: 16%
Test: 80%
``

Reply:
> So, shouldn't the data ratio be like this? Are the sources of train and test data different?

Yes, the sources are different. The train and validation set from mnist_train.csv which contains 40k data points and test comes from mnist_test.csv which has 28k points.

Reply:
Often people use test and validation set interchangeably however they are not the same. Validation set comes from train test split where as test data is data on which we actually use the model we trained. Kindly let me know if this makes sense.

Reply:
Ah, ok. I thought it from the same file :)

Reply:
Should I use validRatio instead I think that makes even more sense.

Reply:
Nice. that makes sense to me as well.

Reply:
Right, making the changed.

Reply:
Done. :)